kernelci/cli.py

```diff
@@ -262,6 +262,18 @@ class Args(object):
         'action': 'store_true'
     }
 
+    data = {
+        'name': '--data',
+        'help': "Data to be submitted to storage",
+        'required': False
+    }
+
+    data_file = {
+        'name': '--data-file',
+        'help': "Path to the file with data to be submitted to storage",
+        'required': False
+    }
```

Comment:
How about just having `--data-file` and reading from `stdin` when using `-`?  I think that would work a bit better than having data part of the command line, so tools that output some test results can directly be piped into kci_data.

Reading from a file is the main use-case so this version is already fine though.

Reply:
Seconding this. Actually, why not have a positional argument for this instead of the option?

I'm not sure if `--data` is required either. If you're using a shell, it's easy enough to submit the data inline with just `kci_data --data-file - <<<'{"foo": "bar"}'`, or better yet just `kci_data <<<'{"foo": "bar"}'`

Or am I missing some use case?

Reply:
I did think of a positional argument, but all the other kci_* command only have options so I thought it was probably best to be consistent.  Maybe we could change that as a follow-up, but being explicit is good for a first stab.

And yes using stdin is useful to do quick experiments on the command line.  It can also be useful for test suites that produce compatible data on their stdout, something like:
```
some-test-suite | kci_data --data-file -
```
But I guess many tests would need to first have the data stored in a file.  Also kci_data is likely to have some sub-commands, and the config needs to be passed as an argument to tell where the data is being sent to.  So a real-world command line would actually look more like this:
```
some-test-suite --output=results.json
kci_data submit --config=api.kernelci.org --token=SECRET --data-file=results.json
```
Past this point, an etherpad probably makes more sense ;)

Reply:
I created mandatory `--data-file` argument to stay consistent with other `kci_*` tools. It expects a path to the data file or `-` to parse `stdin`.